contribute/design/1.x/apis/cstorvolume_policy.md

```diff
@@ -0,0 +1,318 @@
+	Status CStorVolumePolicyStatus `json:"status"`
+}
+
+// CStorVolumePolicySpec ...
+type CStorVolumePolicySpec struct {
+  // Provision represents volume provisioning configuration like replicaAffinity
+  // etc
+	Provision Provision `json:"provision"`
+
+  // TargetSpec represents configuration related to cstor target and its resources
+	Target TargetSpec `json:"target,omitempty"`
+
+  // ReplicaSpec represents configuration related to replicas resources
+	Replica ReplicaSpec `json:"replica,omitempty"`
+
+
+	// ReplicaPoolInfo holds the pool information of volume replicas.
+	// Ex: If volume is provisioned on which CStor pool volume replicas exist
+	ReplicaPoolInfo []ReplicaPoolInfo `json:"replicaPoolInfo,omitempty"`
+}
+
+// TargetSpec represents configuration related to cstor target and its resources
+type TargetSpec struct {
+	// QueueDepth sets the queue size at iSCSI target which limits the
+	// ongoing IO count from client
+	QueueDepth string `json:"queueDepth,omitempty"`
+
+	// Luworkers sets the number of threads that are working on above queue
+	LuWorkers int64 `json:"luWorkers,omitempty"`
+
+	// Monitor enables or disables the target exporter sidecar
+	Monitor bool `json:"monitor,omitempty"`
+
+	// ReplicationFactor represents maximum number of replicas
+	// that are allowed to connect to the target
+	ReplicationFactor int64 `json:"replicationFactor,omitempty"`
+
+	// Resources are the compute resources required by the cstor-target
+	// container.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// AuxResources are the compute resources required by the cstor-target pod
+	// side car containers.
+	AuxResources *corev1.ResourceRequirements `json:"auxResources,omitempty"`
+
+	// Tolerations, if specified, are the target pod's tolerations
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// PodAffinity if specified, are the target pod's affinities
+	PodAffinity *corev1.PodAffinity `json:"affinity,omitempty"`
+
+	// NodeSelector is the labels that will be used to select
+	// a node for target pod scheduleing
+	// Required field
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// PriorityClassName if specified applies to this target pod
+	// If left empty, no priority class is applied.
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+}
+
+// ReplicaSpec represents configuration related to replicas resources
+type ReplicaSpec struct {
+	// ZvolWorkers represents number of threads that executes client IOs
+	ZvolWorkers *int32 `json:"zvolWorkers,omitempty"`
+}
+
+// Provision represents volume provisioning configuration
+type Provision struct {
+	// replicaAffinity is set to true then volume replica resources need to be
+	// distributed across the cstor pool instances based on the given topology
+	ReplicaAffinity bool `json:"replicaAffinity"`
+}
+
+// ReplicaPoolSpec represents the volume replicas pool information
+type ReplicaPoolSpec struct {
```

Comment:
I don't see this being used at all

Reply:
removed , this get renamed to `ReplicaPoolInfo`